File: sdks/typescript/examples/basic_usage.ts
import { AIPClient } from '../src/index';

/**
 * Example: Basic intent resolution
 */
async function main() {
  // Create client (defaults to https://api.jarvisclaw.ai)
  const client = new AIPClient();

  // Example 1: Find cheapest chat completion
  console.log('=== Cheapest Chat Completion ===');
  const costResult = await client.resolve({
    intent: 'chat_completion',
    constraints: { maxPriceUsd: 0.01 },
    preferences: { optimizeFor: 'cost' },
  });
  if (costResult.success && costResult.bestMatch) {
    console.log(`Provider: ${costResult.bestMatch.providerName}`);
    console.log(`Model: ${costResult.bestMatch.model}`);
    console.log(`Price: $${costResult.bestMatch.priceUsd}`);
  }

  // Example 2: Highest quality with function calling
  console.log('\n=== Highest Quality with Function Calling ===');
  const qualityResult = await client.resolve({
    intent: 'chat_completion',
    constraints: { features: ['function_calling'] },
    preferences: { optimizeFor: 'quality' },
  });
  if (qualityResult.success && qualityResult.bestMatch) {
    console.log(`Provider: ${qualityResult.bestMatch.providerName}`);
    console.log(`Score: ${qualityResult.bestMatch.score}`);
  }

  // Example 3: List available providers
  console.log('\n=== Available Providers ===');
  const providers = await client.listProviders('chat_completion');
  providers.forEach(p => {
    console.log(`  ${p.providerName} — ${p.model} ($${p.priceUsd})`);
  });
}

main().catch(console.error);

File: sdks/go/examples/basic_usage.go
// Example usage of the AIP Go SDK.
package main

import (
	"context"
	"fmt"
	"log"

	aip "github.com/jarvisclaw/aip-go"
)

func main() {
	client := aip.NewClient("your-api-key-here")
	ctx := context.Background()

	// ─── Health Check ────────────────────────────────────────
	health, err := client.Health(ctx)
	if err != nil {
		log.Fatalf("health check failed: %v", err)
	}
	fmt.Printf("Platform status: %s (v%s)\n", health.Status, health.Version)

	// ─── Discover available services ─────────────────────────
	catalog, err := client.Discover(ctx, "")
	if err != nil {
		log.Fatalf("discover failed: %v", err)
	}
	fmt.Printf("Available intents: %d\n", len(catalog.Intents))

	// ─── Resolve an intent (without executing) ───────────────
	resolved, err := client.Resolve(ctx, &aip.IntentRequest{
		Intent: "chat",
		Constraints: &aip.Constraints{
			MaxPriceUSD:  0.01,
			MaxLatencyMs: 2000,
		},
		Preferences: &aip.Preferences{
			OptimizeFor: "quality",
		},
	})
	if err != nil {
		log.Fatalf("resolve failed: %v", err)
	}
	fmt.Printf("Best match: %s (%s) — score %.2f\n",
		resolved.BestMatch.ProviderName,
		resolved.BestMatch.Model,
		resolved.BestMatch.Score,
	)

	// ─── Chat (convenience method) ───────────────────────────
	chatResp, err := client.Chat(ctx, &aip.ChatRequest{
		Messages: []aip.ChatMessage{
			{Role: "user", Content: "Explain AIP in one sentence."},
		},
	})
	if err != nil {
		log.Fatalf("chat failed: %v", err)
	}
	fmt.Printf("Reply: %s\n", chatResp.Choices[0].Message.Content)

	// ─── Image generation ────────────────────────────────────
	imgResp, err := client.GenerateImage(ctx, "a futuristic city at sunset", map[string]any{
		"size": "1024x1024",
	})
	if err != nil {
		log.Fatalf("image gen failed: %v", err)
	}
	fmt.Printf("Image URL: %v\n", imgResp["url"])

	// ─── Search ──────────────────────────────────────────────
	searchResp, err := client.Search(ctx, "Agent Intent Protocol", nil)
	if err != nil {
		log.Fatalf("search failed: %v", err)
	}
	fmt.Printf("Search results: %v\n", searchResp["results"])
}
